components/brave_wallet/renderer/brave_wallet_js_handler.cc

```diff
@@ -114,20 +116,32 @@ v8::Local<v8::Promise> BraveWalletJSHandler::Request(v8::Isolate* isolate,
 }
 
 void BraveWalletJSHandler::OnRequest(
-    std::unique_ptr<v8::Global<v8::Promise::Resolver>> promise_resolver,
+    v8::Global<v8::Promise::Resolver> promise_resolver,
     v8::Isolate* isolate,
-    std::unique_ptr<v8::Global<v8::Context>> context_old,
-    const int status,
+    v8::Global<v8::Context> context_old,
+    const int http_code,
     const std::string& response) {
   v8::HandleScope handle_scope(isolate);
-  v8::Local<v8::Context> context = context_old->Get(isolate);
+  v8::Local<v8::Context> context = context_old.Get(isolate);
   v8::Context::Scope context_scope(context);
 
-  v8::Local<v8::Promise::Resolver> resolver = promise_resolver->Get(isolate);
+  v8::Local<v8::Promise::Resolver> resolver = promise_resolver.Get(isolate);
   v8::Local<v8::String> result;
-  result = v8::String::NewFromUtf8(isolate, response.c_str()).ToLocalChecked();
+  bool reject = http_code != 200;
+  ProviderErrors code = ProviderErrors::kDisconnected;
+  std::string message;
+  std::string formed_response;
```

Comment:
Noting that these provider errors are meant to be issued before a request is even issued.  So I think this code will have to move once we have UI to produce the difference kinds of these errors.  We can do it in a later PR.

Reply:
Yeah, I also thought that probably we would need to place it in some common place